code/modules/hydroponics/grown/towercap.dm

```diff
@@ -179,6 +182,48 @@
 	return ..()
 
 /obj/structure/bonfire/attackby(obj/item/W, mob/user, params)
+	//SMOKE SIGNALS. Burn cloth to send a radio message to everyone who knows tribal.
+	if(istype(W, /obj/item/stack/sheet/cloth) && !grill && burning)
+		var/obj/item/stack/sheet/cloth/C = W
```

Comment:
I'd prefer this all to be its own proc, and use more descriptive vars.

Reply:
Will do!